include/openmc/hdf5_interface.h

```diff
@@ -47,36 +47,31 @@ void write_string(hid_t group_id, const char* name, const std::string& buffer,
                   bool indep);
 
 void
-read_nd_vector(hid_t obj_id, const char* name, std::vector<double>& result,
+read_nd_vector(hid_t obj_id, const char* name, xt::xtensor<double, 1>& result,
```

Comment:
The problem with this one is that the compiler can't resolve the functions because xtensor actually has templated parameters `<T, N, L, Tag>`.  I can use a good layout [L], but this `Tag`is an odd unicorn that 1) I can't seem to find a good parameter to use, and 2) seeems related to expressions and therefore I'm not sure it will work generally.

I did, however, remove the xtensor<double, 5> version of read_nd_vector - I no longer need that.

Reply:
I just tried this quickly and got it to work. I think the problem you were seeing was not related to the template parameters but rather the order of declarations. Make sure that the `xtensor` version of `read_dataset` appears *after*` the `xarray` version.

Reply:
Thanks, works now!